shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/domain/model/Question.kt

```diff
@@ -21,43 +20,83 @@ data class Question(
     val coverImageUrl: String,
     val answers: List<Answer>
 ) {
-    private val sortedAnswers: List<Answer>
+    val sortedAnswers: List<Answer>
         get() = answers.sortedBy { it.displayOrder }
-
-    val answerTexts: List<String>
-        get() = sortedAnswers.map { it.text.orEmpty() }
-
-    val answerPlaceholders: List<String>
-        get() = sortedAnswers.map { it.inputMaskPlaceholder.orEmpty() }
 }
 
 sealed class QuestionDisplayType {
 
-    object Intro : QuestionDisplayType()
-    object Star : QuestionDisplayType()
-    object Heart : QuestionDisplayType()
-    object Smiley : QuestionDisplayType()
-    data class Choice(val answers: List<String>) : QuestionDisplayType()
-    object Nps : QuestionDisplayType()
-    data class Textarea(val placeholder: String) : QuestionDisplayType()
-    data class Textfield(val placeholders: List<String>) : QuestionDisplayType()
-    data class Dropdown(val answers: List<String>) : QuestionDisplayType()
-    object Outro : QuestionDisplayType()
-    object Unsupported : QuestionDisplayType()
+    abstract val answers: List<Answerable>
+    abstract var input: List<AnswerInput>
+
+    data class Intro(
+        override val answers: List<Answerable> = emptyList(),
+        override var input: List<AnswerInput> = emptyList()
+    ) : QuestionDisplayType()
+
+    data class Star(
+        override val answers: List<Answerable>,
+        override var input: List<AnswerInput> = emptyList()
+    ) : QuestionDisplayType()
+
+    data class Heart(
+        override val answers: List<Answerable>,
+        override var input: List<AnswerInput> = emptyList()
+    ) : QuestionDisplayType()
```

Comment:
Changing the way to override fields in sealed class could help us avoid having duplicated code, such as `default values`, having more data classes, etc

```
sealed class QuestionDisplayType(
    open val answers: List<Answer> = emptyList(),
    open var input: List<AnswerInput> = emptyList()
) {

    object Intro : QuestionDisplayType()

    data class Star(
        override val answers: List<Answer>
    ) : QuestionDisplayType(answers = answers)

    data class Heart(
        override val answers: List<Answer>,
    ) : QuestionDisplayType(answers = answers)
```

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/123/commits/f972ffa3a07361be38a62957334b6171e77d267d